inductiva/tasks/task.py

```diff
@@ -175,11 +176,35 @@ def wait(self, polling_period: int = 5) -> models.TaskStatusCode:
     def kill(self) -> None:
         """Kill the task.
 
-        This method issues a request to the API to kill the task. It doesn't
-        block waiting for confirmation if the task was killed.
+        This method issues a request to the API to kill the task.
+        Then waits for 5s (checking every second) to see if the task was killed.
+        If the task is still running (or has a status different from killed)
+        after 5s, it issues a warning.
         """
         self._api.kill_task(path_params=self._get_path_params())
 
+        task_status = self.get_status()
+
+        start_time = time.time()
+
+        while task_status == models.TaskStatusCode.PENDINGKILL:
+
+            time.sleep(1)
+            task_status = self.get_status()
+            if time.time() - start_time > constants.MAX_TIME_WAITING_FOR_TASK:
+                logging.warning(
+                    "Task may not have been killed successfully. Task status: "
+                    "%s", task_status)
+                return
+
+        if task_status == models.TaskStatusCode.KILLED:
+            logging.info("Task killed successfully. Task status: %s",
+                         task_status)
```

Comment:
This check needs to be right after the `get_status()` inside the while and if it is killed you break the loop. Otherwise, you will see check the time condition, which can raise a warning even if the status has changed.